sensor/debugger/k8s/trace.go

```diff
@@ -7,39 +7,54 @@ import (
 	"path"
 
 	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/sync"
 )
 
 var _ io.Writer = (*TraceWriter)(nil)
 
 // TraceWriter writes sensor handled k8s events into a file
 type TraceWriter struct {
-	// Destination file where we will store the events
-	Destination string
 	// mu mutex to avoid multiple goroutines writing at the same time
 	mu sync.Mutex
+	f  *os.File
 }
 
-// Init initializes the writer
-func (tw *TraceWriter) Init() error {
-	if path.Dir(tw.Destination) == "" {
-		return nil
+// NewTraceWriter initializes the writer with destination file where we will store the events
+func NewTraceWriter(dest string) (*TraceWriter, error) {
+	if path.Dir(dest) == "" {
+		return nil, errors.New("trace destination must be set")
 	}
-	return os.MkdirAll(path.Dir(tw.Destination), os.ModePerm)
+	err := os.MkdirAll(path.Dir(dest), os.ModePerm)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating trace destination directory")
+	}
+	f, err := os.OpenFile(dest, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
```

Comment:
When is the file closed ? I don't think there is the concept of object destructor in Go, or is there something the garbage collector can call when releasing objects ?
How about a Cleanup function that closes the file and that is deferred where the TraceWriter is instantiated ?

Reply:
I added `Close()` and `defer`ed it in main although the os should handle closing it https://stackoverflow.com/a/10890222